thunder/executors/cudnnex.py

```diff
@@ -354,6 +374,9 @@ def _cudnn_sdpa_checker(
     if cudnn is None:
         return False
 
+    if query.device.type != "cuda" or key.device != query.device or value.device != query.device:
```

Comment:
@t-vi @tfogal the failure in CI was due to cudnnex claiming sdpa for cpu tensors too.
Added a check to guard against it.